### DISCUSSION TITLE
Snapshot restoration from Google Cloud Storage

### DISCUSSION THREAD

**Discussion by BenediktSchuh1324** [qdrant/qdrant#3189] [General]
I am currently working on restoring a snapshot saved on a gcs bucket on a distributed deployment in gke. The easiest way to provide a path to a snapshot would be a gcs uri. I tried it with the 

I'd love for qdrant to support gs:// uris so I can just give the cluster node permission to read from the bucket and use that to recover a snapshot.

It is also possible to host your own webserver to provide the snapshot but that is a lot of work that could be avoided. Another alternative is to allow the use of urls that need authentication. GCS supports the use of authenticated urls. However these need a Authorization Header to work. So currently that is no alternative either.

TLDR:
I want to recover a snapshot from a GCS bucket on a distributed qdrant deployment in GKE

**Comment by timvisee** [comment]
Thank you for the report.

I have no experience with GCS. Is it possible to embed the authentication details in the URL instead, with the `https://user:password@host/` pattern? In that case no changes would be required.

**Comment by generall** [comment]
I think GCS can generate pre-signed url, which should be good enough to me used in qdrant.
